sdk/clientcore/http-stress/src/main/java/io/clientcore/http/stress/util/TelemetryHelper.java

```diff
@@ -160,6 +164,87 @@ public Mono<Void> instrumentRunAsync(Mono<Void> runAsync) {
         });
     }
 
+    /**
+     * Instruments a CompletableFuture: records future duration along with the status (success, error, cancellation),
+     * @param runAsyncFuture the future to instrument
+     * @return the instrumented future
+     */
+    @SuppressWarnings("try")
+    public CompletableFuture<Void> instrumentRunAsyncWithCompletableFuture(CompletableFuture<Void> runAsyncFuture) {
+        Instant start = Instant.now();
+        Span span = tracer.spanBuilder("runAsyncCompletableFuture").startSpan();
+        try (Scope s = span.makeCurrent()) {
+            return runAsyncFuture
+                .whenComplete((result, throwable) -> {
+                    if (throwable != null) {
+                        trackFailure(start, throwable, span);
+                    } else {
+                        trackSuccess(start, span);
+                    }
+                });
+        } finally {
+            span.end();
+        }
```

Comment:
Won't the span end before the task completes? `.whenComplete` doesn't block the return, it's a side-effect operator, so we would end up closing the span before usage of the span finishes.

Should the design change to be more like the ExecutorService where the `whenComplete` tracks ending the span.